### DISCUSSION TITLE
Job Shop Scheduling (Focus on the front part of the makespan)

### DISCUSSION THREAD

**Discussion by ardentchoong** [google/or-tools#3137] [CP-SAT questions]
Hi, I'm very curious about the following problem.

Is there a way to give a larger merit to the front 1/2 of the makespan in case of job shop scheduling with the cp model of ortools?
For example, assuming that there is a task with 4 machines(A, B, C, D), modeling in a way that reduces the latest end time leads to a case where some jobs do not start quickly and may start late. Because it just needs to end earlier than the latest end time.
So if I modify the objective function by reducing the sum of the latest times for each of the 4 machines, there is a problem that some machines do not assign jobs at all.
So, is there a way to make sure that it is more densely placed in the front 1/2 of the makespan? (i.e., let's focus more on the time that's coming soon than the distant future)

**Comment by lperron** [comment]
can you specify the objective ?
Maybe sum of ends of the 4th task of each job ? (or any other subset of tasks)
